### ISSUE TITLE
ct-ng source fails with "Missing 'x86_64-w64-mingw32-ar' or ..."

### ISSUE THREAD

**Issue by archi** [crosstool-ng/crosstool-ng#776]
With `crosstool-ng-1.23.0` on `openSUSE Leap 42.2` (`Linux 4.4.46-11-default x86_64 GNU/Linux`):

I want to download the sources for a bunch of gcc configurations, so I can put all the required sources onto our NFS and disable downloading for our automatic, nightly builds.

Using `ct-ng source` works nice for "normal" builds, but fails for canadian cross.

Error message:

    [ERROR]  Missing: 'x86_64-w64-mingw32-ar' or 'x86_64-w64-mingw32-ar' or 'ar' : either needed!

There is a `ar` at `/usr/bin/ar`, so I suppose that's caused by my normal environment not having a gcc for `x86_64-w64-mingw32`. I believe the `source` is meant for just downloading the sources for later builds, so it should only depend on wget/curl, but not on any actual build tools.

I can use `ct-ng source` to fetch the sources for our linux configurations.

Config files (`powerpc-unknown-eabi` for win64) and build.log are attached in a zip.
Careful, I did not test building a compiler with this config file [waiting for our distributed build system], so if someone stumbles accross this via Google: This is not the configuration file you're looking for ;)
[ct-ng-source-failing.zip](https://github.com/crosstool-ng/crosstool-ng/files/1148683/ct-ng-source-failing.zip).

**Comment by stilor** [comment]
You're right about what `ct-ng source` is supposed to do. But, for historical reasons, ct-ng did the environment checks first. I started reworking the download/patching code in ct-ng, and a first stab is checked in to `master`, but this hasn't been addressed yet. I'll keep this issue open as a reminder.

I won't have much time to devote to ct-ng for the next month or so, though, so don't expect this to change until mid-August.
